xbmc/utils/SaveFileStateJob.cpp

```diff
@@ -35,6 +35,8 @@ void CSaveFileState::DoWork(CFileItem& item,
 
   if (item.HasVideoInfoTag() && StringUtils::StartsWith(item.GetVideoInfoTag()->m_strFileNameAndPath, "removable://"))
     progressTrackingFile = item.GetVideoInfoTag()->m_strFileNameAndPath; // this variable contains removable:// suffixed by disc label+uniqueid or is empty if label not uniquely identified
+  else if (item.HasVideoInfoTag() && item.IsVideoDb())
+    progressTrackingFile = item.GetVideoInfoTag()->m_strFileNameAndPath; // we need the file url of the video db item to create the bookmark
```

Comment:
This could be a potential fix but SaveFileState stores more info than just bookmarks, doesn't it? If this is the case, those other infos can't be accessed anymore.
Another solution could be to use item's path to query bookmarks:
https://github.com/xbmc/xbmc/blob/master/xbmc/video/windows/GUIWindowVideoBase.cpp#L569
That would drop the 7 year old hack done for some reason nobody remembers and aligns DB items with the rest of the code.

Reply:
> This could be a potential fix but SaveFileState stores more info than just bookmarks, doesn't it? If this is the case, those other infos can't be accessed anymore.

 Yes and no, I checked that very careful and all other methods called with `progressTrackingFile` for videos are also expecting `filenenameandpath` instead `videodburl`. So, my fix also fixes the other places. It all turns down to the video database scheme...

> Another solution could be to use item's path to query bookmarks

 Yeah, that's the best approach, but unfortunately this will require a video database change that will break all existing bookmarks (and other stored data, like play count). Not a good idea for v18, but definitely good for v19.

Reply:
ok